presto-main/src/main/java/com/facebook/presto/sql/planner/plan/ApplyNode.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.planner.plan;
+
+import com.facebook.presto.sql.planner.Symbol;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+
+import javax.annotation.concurrent.Immutable;
+
+import java.util.List;
+import java.util.Map;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+@Immutable
+public class ApplyNode
+        extends PlanNode
+{
+    private final PlanNode input;
+    private final PlanNode subquery;
+
+    // Mapping between input and subquery, key represents symbols in input, values in subquery
+    private final Map<Symbol, Symbol> correlation;
```

Comment:
Yes, it is incorrect. I noticed that and I removed it in subsequent commit. Please see: https://github.com/prestodb/presto/pull/4403/commits/d15cf99ca337665758cae48f50c952a85cf198f2#diff-727404f7967058fb13c2898cad1c3784R52